autoload/phpactor/fileutils.vim

```diff
@@ -0,0 +1,47 @@
+function! phpactor#fileutils#searchUpwardBySingleMarker(path, marker) abort
+  let l:markerFile = findfile(a:marker, fnamemodify(a:path, ':p:h').';')
+
+  return empty(l:markerFile) ? v:null : fnamemodify(l:markerFile, ':p:h')
+endfunction
+
+function! phpactor#fileutils#searchDirectoryUpwardByRootMarkers(initialDirectory, workspaceRootMarkers, forbiddenDirs)
+  for l:marker in a:workspaceRootMarkers
+    let l:markerPath = phpactor#fileutils#searchUpwardBySingleMarker(a:initialDirectory, l:marker)
+
+    if v:null is l:markerPath
+      continue
+    endif
+
+    if empty(a:forbiddenDirs) || ! phpactor#fileutils#containsDirectoryFrom(l:markerPath, a:forbiddenDirs)
+      return l:markerPath
+    endif
+  endfor
+
+  return v:null
+endfunction
+
+function! phpactor#fileutils#normalizePath(path) abort
+  let l:modifiers = isdirectory(a:path) ? ':p:h' : ':p'
```

Comment:
Should it not be ?
```suggestion
  let l:modifiers = !isdirectory(a:path) ? ':p:h' : ':p'
```

Otherwise I don't get it.
Also the name is confusing because you're doing things that is not expected, just from the name this is the behavior I expect from such a function:
```viml
echo normalizePath('~/vim/vimrc') " Print: /home/user/vim/vimrc
echo normalizePath('~/vim') " Print: /home/user/vim
```

Reply:
Please wait a while for a vader test that may help you to understand the need to have this function (and eventually propose more accurate name).

Reply:
Please test
```vim
echo fnamemodify('/usr/bin', ':h') " gives /usr
echo fnamemodify('/usr/bin/', ':h') " gives /usr/bin
```

For directories `:p` adds a trailing slash if such did not existed before. We then can use `:h` like in the example above.